components/logins/src/db.rs

```diff
@@ -629,17 +671,20 @@ impl LoginDb {
                 FROM loginsM",
                 changed = SyncStatus::Changed as u8),
             &[(":now_ms", &now_ms as &dyn ToSql)])?;
-
+        scope.err_if_interrupted()?;
+        tx.commit()?;
```

Comment:
general question: does `err_if_interrupted()` rollback the transaction?

Reply:
Yep, more generally, any way we return before reaching `tx.commit()` will roll back the transaction. Note that `?` is roughly like Swift's `try`